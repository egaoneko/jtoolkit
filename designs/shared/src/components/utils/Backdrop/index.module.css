.container {
  @apply absolute inset-0 backdrop-brightness-50 z-auto;
}
