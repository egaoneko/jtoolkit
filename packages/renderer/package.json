{
  "name": "@toolkit/renderer",
  "version": "0.0.1",
  "description": "renderer",
  "main": "lib/index",
  "types": "lib/index",
  "files": [
    "lib"
  ],
  "license": "MIT",
  "scripts": {
    "build": "npm run compile",
    "compile": "tsc --project tsconfig.json",
    "start": "npm run build && node lib/index.js",
    "test": "jest",
    "test:watch": "jest --watch",
    "lint": "eslint \"src/**/*.{js,ts}\" --fix",
    "format": "prettier --write \"src/**/*.ts\""
  },
  "dependencies": {},
  "devDependencies": {}
}