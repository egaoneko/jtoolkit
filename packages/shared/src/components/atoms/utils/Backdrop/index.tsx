import { forwardRef, HTMLAttributes, ReactElement } from 'react';
import clsx from 'clsx';

import styles from './index.module.css';

export type BackdropProps = HTMLAttributes<HTMLDivElement>;

const Backdrop = forwardRef<HTMLDivElement, BackdropProps>(function Backdrop(
  { className, ...props },
  ref,
): ReactElement | null {
  return <div {...props} className={clsx(styles.container, className)} ref={ref} />;
});

export default Backdrop;
