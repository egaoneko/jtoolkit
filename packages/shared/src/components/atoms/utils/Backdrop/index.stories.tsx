import { Story } from '@storybook/react';

import Backdrop from 'src/components/atoms/utils/Backdrop/index';
import styles from 'src/styles/storybook.css';

export default {
  title: 'atoms/utils/Backdrop',
  component: Backdrop,
};

const StandardTemplate: Story = () => {
  return (
    <div className={styles.box}>
      <Backdrop />
    </div>
  );
};

export const Standard = StandardTemplate.bind({});
