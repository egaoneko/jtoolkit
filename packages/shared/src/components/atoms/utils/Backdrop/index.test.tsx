import renderer from 'react-test-renderer';

import Backdrop from 'src/components/atoms/utils/Backdrop/index';

describe('Backdrop Snapshots', () => {
  test('renders Backdrop with default options', async () => {
    const tree = renderer.create(<Backdrop />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders Backdrop with class', async () => {
    const tree = renderer.create(<Backdrop className="test" />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('test');
  });
});
