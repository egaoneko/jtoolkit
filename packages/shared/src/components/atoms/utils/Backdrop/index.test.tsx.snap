// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Snapshots renders Backdrop with class 1`] = `
<div
  className="container test"
/>
`;

exports[`Backdrop Snapshots renders Backdrop with default options 1`] = `
<div
  className="container"
/>
`;
